lib/Agrammon/Model.pm6

```diff
@@ -228,6 +230,14 @@ class Agrammon::Model {
 
         my $evaluator = ModuleRunner.new(:$module, :@dependencies);
         %loaded{$module-name} = $evaluator;
+        %!module-hash{$module-name} = $module;
+        for $module.input -> $input {
+            my @keys;
+            next unless $input.is-filter;
+            for $input.as-hash<enum>.keys -> $key {
+                %!filter-groups{ $module-name ~ '::' ~ $input.name  }{$key} = True;
+            }
+        }
```

Comment:
Hmm, so this seems to be building up a global view of all inputs that can serve as filters across all modules, so it's global in nature.